.github/workflows/generate.yml

```
@@ -0,0 +1,30 @@
+name: generate
+
+on:
+  pull_request:
+
+jobs:
+  generate:
+    name: generate
+    runs-on: ubuntu-latest
+    steps:
+      - name: Collect Metrics
+        id: collect-gha-metrics
+        uses: smartcontractkit/push-gha-metrics-action@v1
+        with:
+          basic-auth: ${{ secrets.GRAFANA_CLOUD_BASIC_AUTH }}
+          hostname: ${{ secrets.GRAFANA_CLOUD_HOST }}
+          this-job-name: generate
+        continue-on-error: true
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - run: make mockery
+      - run: make abigen
+      - run: go mod download
```

Perhaps we should do the same kind of check here after running `make gomodtidy` :thinking: 
```suggestion
      - run: make gomodtidy
      - name: Check Clean
        run: test -n "$(git status --short | tee /dev/tty)" && echo 'Unclean working tree after go mod tidy.' && exit 1
```

---

Added.